packages/cozy-intent/src/api/services/NativeService.ts

```diff
@@ -86,11 +86,27 @@ export class NativeService {
   ): Promise<Connection> =>
     await ParentHandshake(messenger, this.localMethods, numbers.maxAttempts)
 
-  public tryEmit = async (event: NativeEvent): Promise<void> => {
+  public tryEmit = async (
+    event: NativeEvent,
+    componentId: string
```

Comment:
`componentId` will always be defined? It's not marked as optional here. Does it mean every single `webviewIntent.call` necessitate an ID, even if it doesn't touch the UI scope?

Reply:
No, this componentId would be injected by the FlagshipApp here: https://github.com/cozy/cozy-flagship-app/pull/1082/files#diff-94ee99771b8d7d089bb046a665823be8eedf5c5a317115debff9a97a69638ce4R218

This has been done this way so we don't need to update the cozy-apps and retro-compatibility will be ensured

If I understand it correctly, the code I linked is the only place in our entire code base where `tryEmit()` is called

Reply:
but according to the interface https://github.com/cozy/cozy-flagship-app/pull/1082/files#diff-189dde1ff47afcac8c4a195539f67518014f77da3802e819c641838fb7739ebd `componentId` can be `undefined` at some point (not sure about the context, but to me it seems like intended). 
Basically if this variable can be `undefined` I think the cozy-intent should be prepared for it, as in 
```ts
type tryEmit = (event: NativeEvent, componentId?: string) => Promise<void>
```

Reply:
This was an error on Flagship App side. As this is always injected from `tryEmit`, then it cannot be undefined on Flagship App side.

Fixed in https://github.com/cozy/cozy-flagship-app/compare/4bb373b0bb797f98fd9ad1c915466c499c38a22f..7b2a6c9394ef67df5b5b4b74b361742eb5bd9872